x/gmp/module_ibc.go

```diff
@@ -151,26 +151,29 @@ func (im IBCModule) OnRecvPacket(
 		ctx.Logger().Debug(fmt.Sprintf("Got general message with token: %v", msg))
 		payloadType, err := abi.NewType("string", "", nil)
 		if err != nil {
-			ctx.Logger().Info(fmt.Sprintf("failed to create reflection: %s", err.Error()))
+			ctx.Logger().Error(fmt.Sprintf("failed to create reflection: %s", err.Error()))
 			return channeltypes.NewErrorAcknowledgement(cosmossdkerrors.Wrapf(transfertypes.ErrInvalidMemo, "unable to define new abi type (%s)", err.Error()))
 		}
 		payloadData := msg.Payload
 		ctx.Logger().Debug(fmt.Sprintf("ABI-encoded data: %v", hex.EncodeToString(payloadData)))
 
 		args, err := abi.Arguments{{Type: payloadType}}.Unpack(payloadData)
 		if err != nil {
+			// unpack can fail for payloads like "{}", in this case we just forward the packet
 			ctx.Logger().Debug(fmt.Sprintf("failed to unpack: %s", err.Error()))
-			return channeltypes.NewErrorAcknowledgement(cosmossdkerrors.Wrapf(transfertypes.ErrInvalidMemo, "unable to unpack payload (%s)", err.Error()))
+			return im.app.OnRecvPacket(ctx, modulePacket, relayer)
 		}
 		pfmPayload := args[0].(string)
 		data.Memo = string(pfmPayload)
```

Comment:
this looks weird, we're casting to `string` and then call `string(...)`; seems duplicated